digdag-core/src/main/java/io/digdag/core/agent/LocalAgentManager.java

```diff
@@ -15,8 +15,8 @@
         implements BackgroundExecutor
 {
     private final Supplier<MultiThreadAgent> agentFactory;
-    private Thread thread;
-    private MultiThreadAgent agent;
+    private volatile Thread thread;
```

Comment:
[minor] This keyword doesn't seem to be needed because this value is called only in `synchronized` method.